jwst/extract_1d/extract.py

```diff
@@ -473,3159 +377,1165 @@ def log_initial_parameters(extract_params):
+        spec.source_name = slit.source_name
 
-    Extended summary
-    ----------------
-    The more restrictive (i.e. larger) limit is the one upon which the
-    output value will be based.  If the WCS limit is larger, the value will
-    be increased to an integer, on the assumption that WCS lower limits
-    correspond to the lower edge of the bounding pixel.  If this value
-    will actually be used for an extraction limit (i.e. if the limits were
-    not already specified by polynomial coefficients), then
-    `assign_polynomial_limits` will create a polynomial function using this
-    value, except that it will be decreased by 0.5 to correspond to the
-    lower edge of the bounding pixels.
+    if hasattr(slit, "source_alias") and slit.source_alias is not None:
+        spec.source_alias = slit.source_alias
 
-    Parameters
-    ----------
-    aperture_start : int or float
-        xstart or ystart, as specified by the extract1d reference file or the image
-        size.
+    if hasattr(slit, "source_type") and slit.source_type is not None:
+        spec.source_type = slit.source_type
 
-    wcs_bb_lower_lim : int or float
-        The lower limit from the WCS bounding box.
+    if hasattr(slit, "stellarity") and slit.stellarity is not None:
+        spec.stellarity = slit.stellarity
 
-    Returns
-    -------
-    value : int or float
-        The start limit, possibly constrained by the WCS start limit.
-    """
-    if aperture_start >= wcs_bb_lower_lim:  # ref is inside WCS limit
-        return aperture_start
-    else:  # outside (below) WCS limit
-        return math.ceil(wcs_bb_lower_lim)
+    if hasattr(slit, "source_xpos"):
+        spec.source_xpos = slit.source_xpos
 
+    if hasattr(slit, "source_ypos"):
+        spec.source_ypos = slit.source_ypos
 
-def compare_stop(aperture_stop, wcs_bb_upper_lim):
-    """Compare the stop limit from the aperture with the WCS upper limit.
+    if hasattr(slit, "source_ra"):
+        spec.source_ra = slit.source_ra
 
-    The more restrictive (i.e. smaller) limit is the one upon which the
-    output value will be based.  If the WCS limit is smaller, the value
-    will be truncated to an integer, on the assumption that WCS upper
-    limits correspond to the upper edge of the bounding pixel.  If this
-    value will actually be used for an extraction limit (i.e. if the
-    limits were not already specified by polynomial coefficients), then
-    `assign_polynomial_limits` will create a polynomial function using this
-    value, except that it will be increased by 0.5 to correspond to the
-    upper edge of the bounding pixels.
+    if hasattr(slit, "source_dec"):
+        spec.source_dec = slit.source_dec
 
-    Parameters
-    ----------
-    aperture_stop : int or float
-        xstop or ystop, as specified by the extract1d reference file or the image
-        size.
+    if hasattr(slit, "shutter_state"):
+        spec.shutter_state = slit.shutter_state
```

Comment:
Question: there are a lot of very similar lines here with strings/attributes written three times each.  Is the code below preferable for setting everything but slitname?

```
for varstring in ["slitlet_id", "source_id", "source_xpos", "source_ypos",
                  "source_ra", "source_dec", "shutter_state"]:
    if hasattr(slit, varstring):
        setattr(spec, varstring, getattr(slit, varstring))

for varstring in ["source_name", "source_alias", "source_type", "stellarity"]:
    if getattr(slit, varstring, None) is not None:
        setattr(spec, varstring, getattr(slit, varstring))
```

Reply:
Sure, that saves a few lines of code and is easily expandable.  I'll make this change.